RFS/src/main/java/com/rfs/version_os_2_11/IndexCreator_OS_2_11.java

```diff
@@ -41,6 +44,48 @@ public Optional<ObjectNode> create(
         body.set("settings", settings);
 
         // Create the index; it's fine if it already exists
-        return client.createIndex(indexName, body, context);
+        try {
+            return client.createIndex(indexName, body, context);
+        } catch (InvalidResponse invalidResponse) {
+            var illegalArguments = invalidResponse.getIllegalArguments();
+
+            if (illegalArguments.isEmpty()) {
+                log.debug("Cannot retry invalid response.");
+                return Optional.empty();
+            }
+
+            for (var illegalArgument : illegalArguments) {
+                if (!illegalArgument.startsWith("index.")) {
+                    log.warn("Expecting all retryable errors to start with 'index.', instead saw " + illegalArgument);
+                    return Optional.empty();
+                }
+
+                var shortenedIllegalArgument = illegalArgument.replaceFirst("index.", "");
+                removeFieldsByPath(settings, shortenedIllegalArgument);
+            }
+
+            log.debug("Reattempting creation of index after removing illegal arguments; " + illegalArguments);
+            return client.createIndex(indexName, body, context);
+        }
+    }
+
+    private void removeFieldsByPath(ObjectNode node, String path) {
+        var pathParts = path.split("\\.");
+
+        if (pathParts.length == 1) {
+            node.remove(pathParts[0]);
+            return;
+        }
+
+        var currentNode = node;
+        for (int i = 0; i < pathParts.length - 1; i++) {
+            var nextNode = currentNode.get(pathParts[i]);
+            if (nextNode != null && nextNode.isObject()) {
+                currentNode = (ObjectNode) nextNode;
+            } else {
+                return;
+            }
+        }
+        currentNode.remove(pathParts[pathParts.length - 1]);
```

Comment:
Double checking, but I assume this handles string fields/objects whatever because we're just removing the whole node. Correct?

Reply:
Yup - both casss are handled.

Reply:
- Created unit tests that prove this statement https://github.com/opensearch-project/opensearch-migrations/pull/881